api/src/controllers/users.ts

```diff
@@ -15,6 +16,54 @@ router.get('/', function (req, res, next) {
   res.json(req.session);
 });
 
+router.get('/preferences', (req, res) => {
+  if (!req.session.passport) {
+    res.json({ error: 'Must be logged in to get preferences.' });
+    return;
+  }
+
+  const userID = req.session.passport.user.id;
+
+  getDocuments(COLLECTION_NAMES.PREFERENCES, { _id: userID }).then((preferences) => {
+    if (preferences.length > 0) {
+      res.json(preferences[0]);
+    } else {
+      res.json({ error: 'No preferences found' });
+    }
+  });
+});
```

Comment:
Why not use an async function for the request handler?

```suggestion
router.get('/preferences', async (req, res) => {
  if (!req.session.passport) {
    res.json({ error: 'Must be logged in to get preferences.' });
    return;
  }

  const userID = req.session.passport.user.id;

  const preferences = getDocuments(COLLECTION_NAMES.PREFERENCES, { _id: userID });
  if (preferences.length > 0) return res.json(preferences[0]);
  res.json({ error: 'No preferences found' });
});
```

Reply:
Good question